src/test-helpers.tsx

```diff
@@ -393,6 +394,22 @@ export const createFakeLinterMessagesByPath = ({
   return map[path];
 };
 
+export const createFakeCompareInfo = ({
```

Comment:
It will be a bit more verbose but this should hook into `createInternalCompareInfo()` to mimic real world usage as much as possible (patch below). 

We need to be careful about manually constructing internal objects in tests since it could be fragile. It's safer to start with a known external object, i.e. one that would come from our API.

```diff
diff --git a/src/reducers/versions.tsx b/src/reducers/versions.tsx
index c339f6c..990b9f8 100644
--- a/src/reducers/versions.tsx
+++ b/src/reducers/versions.tsx
@@ -166,7 +166,7 @@ export type VersionFile = {
   version: string;
 };
 
-type VersionEntry = {
+export type VersionEntry = {
   depth: number;
   filename: string;
   mimeType: string;
diff --git a/src/test-helpers.tsx b/src/test-helpers.tsx
index edc737c..3009cb5 100644
--- a/src/test-helpers.tsx
+++ b/src/test-helpers.tsx
@@ -26,8 +26,10 @@ import {
   ExternalVersionsList,
   ExternalVersionsListItem,
   Version,
+  VersionEntry,
   VersionEntryType,
   actions as versionsActions,
+  createInternalCompareInfo,
   createInternalDiff,
   createInternalVersion,
   createInternalVersionEntry,
@@ -398,16 +400,24 @@ export const createFakeCompareInfo = ({
   baseVersionId = 1,
   headVersionId = 2,
   mimeType = 'mime/type',
+  entry = createInternalVersionEntry({
+    ...fakeVersionEntry,
+    mimetype: mimeType,
+  }),
   version = fakeVersionWithDiff,
+}: {
+  mimeType?: string;
+  baseVersionId?: number;
+  headVersionId?: number;
+  entry?: VersionEntry;
+  version?: ExternalVersionWithDiff;
 } = {}) => {
-  return {
-    diff: createInternalDiff({
-      baseVersionId,
-      headVersionId,
-      version,
-    }),
-    mimeType,
-  };
+  return createInternalCompareInfo({
+    baseVersionId,
+    entry,
+    headVersionId,
+    version,
+  });
 };
 
 export const createFakeLocation = (props = {}): Location => {
```